src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -597,7 +597,11 @@ export class ListDatabasesActionItem extends Disposable implements IActionViewIt
 
 		// Allows database selector to commit typed or pasted DB names without the need to click
 		// or press enter to make a selection when focus is moved away from the selector.
-		this._register(this._dropdown.onBlur(() => this.databaseSelected(this._dropdown.value)));
+		this._register(this._dropdown.onBlur(() => {
+			if (this._dropdown.value !== this.getCurrentDatabaseName()) {
```

Comment:
We should be doing this check inside databaseSelected instead so we avoid switching the DB unless the db actually changed in all scenarios.